app/assets/javascripts/mobile/mobile_drawer.js

```diff
@@ -1,20 +1,147 @@
 (function(){
   Diaspora.Mobile.Drawer = {
     initialize: function(){
+      this.drawer = $("#drawer");
+      this.mainContent = $("#main");
+      this.dragZone = this.createDragZone();
+      this.mainContent.append(this.dragZone);
+
+      var self = this;
+      var dateLastEvent = Date.now();
+      var debouncingThreshold = 40;
+      var hammerOpts = {
+        "prevent_default": false,
+        threshold: 0,
+        direction: Hammer.DIRECTION_HORIZONTAL
+      };
+
       $("#all_aspects").bind("tap click", function(evt){
         evt.preventDefault();
         $(this).find("+ li").toggleClass("hide");
       });
 
-      $("#menu-badge").bind("tap click", function(evt){
+      $("#followed_tags").bind("tap click", function(evt){
         evt.preventDefault();
-        $("#app").toggleClass("draw");
+        $(this).find("+ li").toggleClass("hide");
       });
 
-      $("#followed_tags").bind("tap click", function(evt){
+      $("#menu-badge").on("tap click", function(evt){
         evt.preventDefault();
-        $(this).find("+ li").toggleClass("hide");
+        self.createOverlay();
+        self.openDrawer();
       });
+
+      this.mainContent.on("tap click", function(){
+        if(self.isDrawerOpened()){
+          self.closeDrawer();
+        }
+      });
+
+      this.dragZone.hammer(hammerOpts).bind("panstart", function(){
+          self.createOverlay();
```

Comment:
- Expected to return a value at the end of this function.
- Missing space before opening brace.